openml/flows/flow.py

```diff
@@ -338,9 +338,24 @@ def publish(self):
         file_elements = {'description': xml_description}
         return_value = _perform_api_call("flow/", file_elements=file_elements)
         self.flow_id = int(xmltodict.parse(return_value)['oml:upload_flow']['oml:id'])
+        try:
+            _check_flow(self)
+        except ValueError as e:
+            message = e.args[0]
+            raise ValueError("Flow was not stored correctly on the server. "
+                             "New flow ID is %d. Please check manually and "
+                             "remove the flow if necessary! Error is:\n'%s'" %
+                             (self.flow_id, message))
         return self
 
 
 def _add_if_nonempty(dic, key, value):
     if value is not None:
         dic[key] = value
+
+
+def _check_flow(flow):
+    import openml.flows.functions
```

Comment:
Importing at the top won't work here due to cyclic imports. I'll add a comment mentioning this.